spring-vault-core/src/main/java/org/springframework/vault/core/lease/SecretLeaseContainer.java

```diff
@@ -375,7 +385,7 @@ public void destroy() throws Exception {
 					Lease lease = entry.getValue().getLease();
 					entry.getValue().disableScheduleRenewal();
 
-					if (lease != null) {
+					if (lease != null && !lease.isRotatingGenericLease()) {
```

Comment:
Leases without a lease Id aren't required to be revoked because there's no Lease Id. `Lease.none` is as well a candidate that does not require a revocation. Call it a bug fix to be truly honest.